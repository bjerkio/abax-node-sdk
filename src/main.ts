export * from './abax-client.js';
export * from './calls/get-odometer-values.js';
export * from './calls/list-trip-expenses.js';
export * from './calls/list-trips.js';
export * from './calls/list-vehicles.js';
export * from './calls/shared.js';
export * from './calls/list-equipment-logs.js';
export * from './calls/get-equipment.js';
export * from './calls/list-equipment.js';
export * from './authentication/abax-auth.js';
export * from './authentication/types.js';
export type { AuthorizationCodeInput } from './authentication/calls.js';
