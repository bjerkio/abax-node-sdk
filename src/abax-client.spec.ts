import dotenv from 'dotenv';
import { describe, expect, it } from 'vitest';
import { AbaxClient, Scope, authenticateWithClientCredentials } from './main';
dotenv.config();

describe('abax-client', () => {
  const initialiseClient = async () => {
    const clientId = process.env['ABAX_CLIENT_ID'] ?? 'no-client-id';
    const clientSecret =
      process.env['ABAX_CLIENT_SECRET'] ?? 'no-client-secret';

    const authReply = await authenticateWithClientCredentials({
      clientId,
      clientSecret,
      authUrl: 'https://identity.abax.cloud',
      scopes: [Scope.OpenApi, Scope.OpenApiEquipment],
    });

    return new AbaxClient({
      baseUrl: 'production',
      apiKey: authReply.access_token,
    });
  };
  it('should list equipment', async () => {
    const client = await initialiseClient();

    const result = await client.listEquipment({ page: 1, page_size: 2 });

    expect(result).toMatchSnapshot();
  });

  it('should get equipment', async () => {
    const client = await initialiseClient();

    const result = await client.getEquipment({
      id: '9107a2942ece5b40a642fe2e4c9a84d4',
    });

    expect(result).toMatchSnapshot();
  });
});
