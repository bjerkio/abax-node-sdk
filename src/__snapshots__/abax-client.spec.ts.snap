// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`abax-client > should get equipment 1`] = `
{
  "alias": "3254",
  "asset_id": "3627af59e6f449fb83df4df327925a03",
  "id": "9107a2942ece5b40a642fe2e4c9a84d4",
  "location": {
    "accuracy_radius": 30,
    "in_movement": false,
    "latitude": 59.10668,
    "longitude": 10.102647,
    "signal_source": "Gps",
    "speed": 0,
    "timestamp": "2022-09-08T08:38:52.000+00:00",
  },
  "model": {
    "name": "ZX250LC-7",
  },
  "notes": "HITACHI",
  "organization": {
    "id": "844e8098bb7c58efb3c9c4c6e711a64c",
    "name": "Product Beta NO",
  },
  "unit": {
    "health": "Healthy",
    "id": "c83db34e122353ac9110ee44cd842c34",
    "status": "Deactivated",
    "type": "Unknown",
  },
}
`;

exports[`abax-client > should list equipment 1`] = `
{
  "items": [
    {
      "alias": "3254",
      "asset_id": "3627af59e6f449fb83df4df327925a03",
      "id": "9107a2942ece5b40a642fe2e4c9a84d4",
      "location": {
        "accuracy_radius": 30,
        "in_movement": false,
        "latitude": 59.10668,
        "longitude": 10.102647,
        "signal_source": "Gps",
        "speed": 0,
        "timestamp": "2022-09-08T08:38:52.000+00:00",
      },
      "model": {
        "name": "ZX250LC-7",
      },
      "notes": "HITACHI",
      "organization": {
        "id": "844e8098bb7c58efb3c9c4c6e711a64c",
        "name": "Product Beta NO",
      },
      "unit": {
        "health": "Healthy",
        "id": "c83db34e122353ac9110ee44cd842c34",
        "status": "Deactivated",
        "type": "Unknown",
      },
    },
    {
      "alias": "3222",
      "asset_id": "42f746a35c244c4bbf26876b56f862d9",
      "id": "5621be6961f35661a900748846d2fc31",
      "location": {
        "accuracy_radius": 30,
        "in_movement": false,
        "latitude": 59.90408,
        "longitude": 10.602942,
        "signal_source": "Gps",
        "speed": 0,
        "timestamp": "2022-06-14T16:49:36.000+00:00",
      },
      "model": {
        "name": "ZX350-6",
      },
      "notes": "HITACHI",
      "organization": {
        "id": "844e8098bb7c58efb3c9c4c6e711a64c",
        "name": "Product Beta NO",
      },
      "unit": {
        "health": "Healthy",
        "id": "45f872d43abb5b2fb08619857b045170",
        "status": "Deactivated",
        "type": "Unknown",
      },
    },
  ],
  "page": 1,
  "page_size": 2,
}
`;
